src/backend/mssql.rs

```diff
@@ -0,0 +1,180 @@
+        format!("CREATE TABLE {}[{}]", quoted_prefix!(schema), name)
+    }
+
+    fn create_table_if_not_exists(name: &str, schema: Option<&str>) -> String {
+        let table = format!("{}[{}]", quoted_prefix!(schema), name);
+
+        format!(
+            "IF NOT EXISTS (SELECT * FROM sysobjects WHERE name='{table}' AND xtype='U') CREATE TABLE {quoted}",
+            table = name,
+            quoted = table,
+        )
+    }
+
+    fn drop_table(name: &str, schema: Option<&str>) -> String {
+        format!("DROP TABLE {}[{}]", quoted_prefix!(schema), name)
+    }
+
+    fn drop_table_if_exists(name: &str, schema: Option<&str>) -> String {
+        format!("DROP TABLE IF EXISTS {}[{}]", quoted_prefix!(schema), name)
+    }
+
+    fn rename_table(old: &str, new: &str, schema: Option<&str>) -> String {
+        let prefix = prefix!(schema);
+        format!(r#"EXEC sp_rename '{}{}', '{}'"#, prefix, old, new)
+    }
+
+    fn alter_table(name: &str, schema: Option<&str>) -> String {
+        format!("ALTER TABLE {}[{}]", quoted_prefix!(schema), name)
+    }
+
+    fn add_column(ex: bool, schema: Option<&str>, name: &str, tt: &Type) -> String {
+        let bt: BaseType = tt.get_inner();
+        use self::BaseType::*;
+
+        #[cfg_attr(rustfmt, rustfmt_skip)] /* This shouldn't be formatted. It's too long */
+        format!(
+            "{}{}{}{}{}",
+            match bt {
+                Text => format!("{}[{}] {}", MsSql::prefix(ex), name, MsSql::print_type(bt, schema)),
+                Varchar(_) => format!("{}[{}] {}", MsSql::prefix(ex), name, MsSql::print_type(bt, schema)),
+                Primary => format!("{}[{}] {}", MsSql::prefix(ex), name, MsSql::print_type(bt, schema)),
+                Integer => format!("{}[{}] {}", MsSql::prefix(ex), name, MsSql::print_type(bt, schema)),
+                Float => format!("{}[{}] {}", MsSql::prefix(ex), name, MsSql::print_type(bt, schema)),
+                Double => format!("{}[{}] {}", MsSql::prefix(ex), name, MsSql::print_type(bt, schema)),
+                UUID => format!("{}[{}] {}", MsSql::prefix(ex), name, MsSql::print_type(bt, schema)),
+                Json => format!("{}[{}] {}", MsSql::prefix(ex), name, MsSql::print_type(bt, schema)),
+                Boolean => format!("{}[{}] {}", MsSql::prefix(ex), name, MsSql::print_type(bt, schema)),
+                Date => format!("{}[{}] {}", MsSql::prefix(ex), name, MsSql::print_type(bt, schema)),
+                Binary => format!("{}[{}] {}", MsSql::prefix(ex), name, MsSql::print_type(bt, schema)),
+                Foreign(_, _, _) => format!("{}[{}] {}", MsSql::prefix(ex), name, MsSql::print_type(bt, schema)),
+                Custom(_) => format!("{}[{}] {}", MsSql::prefix(ex), name, MsSql::print_type(bt, schema)),
+                Array(_) => panic!("Arrays are not supported with SQL Server"),
+                Index(_) => unreachable!(), // Indices are handled via custom builder
```

Comment:
Maybe adding the note inside the `unreachable` statement is more descriptive if we ever hit this. Also we should probably pick either `panic` or `unreachable` here: they both lower to the same code so keeping them consistent is good.
```suggestion
                Index(_) => unreachable!("Indices are handled via custom builder"),
```